### PR TITLE
Add consultation schema and examples

### PR DESCRIPTION
**Pull request by fofr**
* Add details and links schema for consultations
* Add examples of:
  * Closed consultation
  * Closed consultation with outcome
  * Closed consultation with outcome and public feedback
  * Open consultation
  * Open consultation with ways to respond
  * Unopened consultation with open date set in the future

There are no known features missing from the schema.

Examples are rendered with this PR: https://github.com/alphagov/government-frontend/pull/201

Paired with @binaryberry 

Story: https://trello.com/c/QseqtCaJ/509-consultations-migration-basic-content-schema-examples-and-front-end-work-large

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -6,3 +6,7 @@ gem 'rake'
 gem 'rspec'
 gem 'govuk-dummy_content_store', '0.0.5'
 gem 'foreman', '0.78.0'
+
+group :test do
+  gem 'pry-byebug'
```
Comment (andrewgarner) [inline Gemfile]: Maybe cherry-pick this out into its own PR as isn't related directly to this group of changes?

File: formats/consultation/frontend/examples/closed_consultation.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  "updated_at": "2016-09-05T15:00:00Z",
+  "schema_name": "consultation",
+  "document_type": "closed_consultation",
+  "locale": "en",
+  "details": {
+    "body": "<div class=\"govspeak\"><p>The Museums Review will consider how museums and galleries across England can thrive and become even more inclusive. The Review will look at how the sector operates, the challenges it faces and the role of government-sponsored museums.</p>\n\n<p>To help inform this review, DCMS would welcome your feedback through our online Calls for Evidence (see links in attached document).</p>\n\n<p>The Calls for Evidence are open until 31 October 2016.</p>\n\n<p>The full Terms of Reference for the review can be found in the above document.</p>\n\n<p>If you have any queries about the review, please contact DCMS at museums.review@culture.gov.uk</p>\n</div>",
+    "first_public_at": "2016-09-05T14:00:00+01:00",
+    "opening_date": "2016-09-05T14:00:00+01:00",
+    "closing_date": "2016-10-31T17:00:00+01:00",
+    "change_history": [
```
Comment (andrewgarner) [inline formats/consultation/frontend/examples/closed_consultation.json]: Only `body`, `government` and `political` are required attributes of `details` so we shouldn't expect to see an empty `change_history`.

File: formats/consultation/frontend/examples/closed_consultation.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    ],
+    "tags": {
+      "browse_pages": [],
```
Comment (andrewgarner) [inline formats/consultation/frontend/examples/closed_consultation.json]: Again `browse_pages` are not required so should we include an empty attribute?

File: formats/consultation/frontend/examples/consultation_outcome.json [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    ],
+    "tags": {
```
Comment (andrewgarner) [inline formats/consultation/frontend/examples/consultation_outcome.json]: Again, these are optional according to the schema.

File: formats/consultation/frontend/examples/closed_consultation.json (hunk 2 above)
Comment (fofr) [inline formats/consultation/frontend/examples/closed_consultation.json]: This is an interesting one. Removing `change_history` causes the frontend to break. Which means the frontend should be more resilient.

File: formats/consultation/frontend/examples/closed_consultation.json (hunk 3 above)
Comment (fofr) [inline formats/consultation/frontend/examples/closed_consultation.json]: There's hidden complexity here: https://github.com/alphagov/govuk-content-schemas/pull/292

File: Gemfile (hunk 1 above)
Comment (binaryberry) [inline Gemfile]: In other GOV.UK repos I've found both examples of this being done within [a dedicated PR](https://github.com/alphagov/publishing-api/pull/114), and examples of it being done within a [PR that was doing something else](https://github.com/alphagov/frontend/pull/1004/commits/854751da9542dc47b24eb9969cadc6d966c71a08). 
PRs create a fair bit of noise (github notifications) and overhead (additional item to deploy), which I feel are not worth having for just adding a debugging gem. IMO avoiding this is more important than having a commit not directly related to the PR here. 
That being said, it's a matter of opinion and dedicated PRs have happened before, so if one of you has strong feelings about it, I can remove it to not block this PR.

File: formats/consultation/frontend/examples/closed_consultation.json (hunk 2 above)
Comment (fofr) [inline formats/consultation/frontend/examples/closed_consultation.json]: Fixed in f7981a81832a10df11d7e0dbfb27970fc8248b52

File: Gemfile (hunk 1 above)
Comment (andrewgarner) [inline Gemfile]: One benefit would be that we could easily revert our changes without unintentionally causing a side effect to anything else that subsequently uses byebug.

Comment (andrewgarner) [inline Gemfile]: One benefit would be that we could easily revert our changes without unintentionally causing a side effect to anything else that subsequently uses byebug.

Comment (binaryberry) [inline Gemfile]: Fair enough - now removed that commit from this PR and opened #446.

File: formats/consultation/frontend/examples/closed_consultation.json [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  "document_type": "closed_consultation",
+  "locale": "en",
+  "details": {
+    "body": "<div class=\"govspeak\"><p>The Museums Review will consider how museums and galleries across England can thrive and become even more inclusive. The Review will look at how the sector operates, the challenges it faces and the role of government-sponsored museums.</p>\n\n<p>To help inform this review, DCMS would welcome your feedback through our online Calls for Evidence (see links in attached document).</p>\n\n<p>The Calls for Evidence are open until 31 October 2016.</p>\n\n<p>The full Terms of Reference for the review can be found in the above document.</p>\n\n<p>If you have any queries about the review, please contact DCMS at museums.review@culture.gov.uk</p>\n</div>",
+    "first_public_at": "2016-09-05T14:00:00+01:00",
+    "opening_date": "2016-09-05T14:00:00+01:00",
+    "closing_date": "2016-10-31T17:00:00+01:00",
+    "tags": {
+      "browse_pages": [],
+      "topics": ["5d37acc4-7631-11e4-a3cb-005056011aef"],
```
Comment (gpeng) [inline formats/consultation/frontend/examples/closed_consultation.json]: I don't think these are required here but they (topics and policies) should be in the links. email-alert-api checks both https://github.com/alphagov/email-alert-api/blob/master/app/workers/notification_worker.rb#L40-L49

Comment (fofr) [inline formats/consultation/frontend/examples/closed_consultation.json]: Topics and policy areas are included in the base links:
https://github.com/alphagov/govuk-content-schemas/blob/master/formats/base_links.json

And policies have been added to `links.json`:
https://github.com/alphagov/govuk-content-schemas/pull/445/commits/77d4b9fecef6bda864b2cb13c69d6a8fc76ef7df#diff-15d8cbec358c915366ee6cbc4bd4e147R6

Comment (fofr) [inline formats/consultation/frontend/examples/closed_consultation.json]: I believe this line was copied directly from the live content item placeholder.

**Comment by fofr** [comment]
This will need a rebase and new examples run against this: https://github.com/alphagov/govuk-content-schemas/pull/448

File: formats/consultation/frontend/examples/closed_consultation.json (hunk 5 above)
Comment (gpeng) [inline formats/consultation/frontend/examples/closed_consultation.json]: Discussed offline... There is another example containing `topics` and `policies` links. We are going to leave the `tags` in the examples for now and test removing them from the presented item in Whitehall.

**Comment by andrewgarner** [comment]
I'll rebase @fofr

**Pull request merged**